### ISSUE TITLE
can: stm32: Build issue on g4 target

### ISSUE THREAD

**Issue by erwango** [zephyrproject-rtos/zephyr#43899]
**Describe the bug**
`twister -b -p nucleo_g474re  -T samples/drivers/can/sample.drivers.can`

```
/__w/zephyr/zephyr/drivers/can/can_stm32fd.c
/__w/zephyr/zephyr/drivers/can/can_stm32fd.c: In function 'can_stm32fd_get_max_bitrate':
/__w/zephyr/zephyr/drivers/can/can_stm32fd.c:161:20: error: 'const struct can_stm32fd_config' has no member named 'mcan'
  161 |  *max_bitrate = cfg->mcan.max_bitrate;
      |                    ^~
/__w/zephyr/zephyr/drivers/can/can_stm32fd.c: In function 'can_stm32fd_line_1_isr':
/__w/zephyr/zephyr/drivers/can/can_stm32fd.c:181:37: error: 'data' undeclared (first use in this function)
  181 |  struct can_mcan_data *mcan_data = &data->mcan_data;
      |                                     ^~~~
```
**Impact**
CI blocked

**Comment by erwango** [comment]
@alexanderwachter Would you  have time to have a look ?

**Comment by henrikbrixandersen** [comment]
@erwango My apologies. Fix is in #43901.
